### PR TITLE
Add `include` param support to GitHub Data Connector

### PR DESCRIPTION
**Pull request by sgrebnov**
## 🗣 Description

Add param to specify which files to include for the GitHub Data Connector.

```
- from: github:github.com/spiceai/spiceai/files/trunk
  name: spiceai.files
  params:
    github_access_token: ${secrets:github_token}
    include: "*.json;*/ISSUE_TEMPLATE/*"
```

```shell
sql> select path from spiceai.files
+----------------------------------------------------------+
| path                                                     |
+----------------------------------------------------------+
| .github/ISSUE_TEMPLATE/bug_report.md                     |
| .github/ISSUE_TEMPLATE/end_game.md                       |
| .github/ISSUE_TEMPLATE/enhancement.md                    |
| .github/ISSUE_TEMPLATE/feature_request.md                |
| .schema/spicepod.schema.json                             |
| .vscode/launch.json                                      |
| crates/data_components/src/debezium/tests/all_types.json |
| crates/llms/src/chat/template.json                       |
| monitoring/datadog-dashboard.json                        |
| monitoring/grafana-dashboard.json                        |
+----------------------------------------------------------+
```

## 🔨 Related Issues

https://github.com/spiceai/spiceai/issues/2384

## 🤔 Concerns

~Ideally we would like to have param as `include` instead of `github_include`, but it seems all params must be prefixed with data connector name.~

### CODE REVIEW

File: crates/data_components/src/github.rs [hunk 1]
```diff
@@ -122,7 +126,8 @@ impl TableProvider for GithubFilesTableProvider {
                 &self.owner,
                 &self.repo,
                 &self.tree_sha,
-                limit,
+                None,
```
Comment (sgrebnov) [inline crates/data_components/src/github.rs:129]: No need for filtering during API check

File: crates/data_components/src/github.rs [hunk 2]
```diff
@@ -72,7 +75,7 @@ impl GithubFilesTableProvider {
 
         // ensure configuration is correct
         client
-            .fetch_files(owner, repo, tree_sha, Some(1), Arc::clone(&schema))
+            .fetch_files(owner, repo, tree_sha, Some(1), None, Arc::clone(&schema))
```
Comment (sgrebnov) [inline crates/data_components/src/github.rs:78]: No need for filtering (`include`) during API check. Limit 1 is used instead

File: crates/data_components/src/github.rs [hunk 3]
```diff
@@ -156,18 +161,24 @@ impl GithubRestClient {
         repo: &str,
         tree_sha: &str,
         limit: Option<usize>,
+        include_patern: Option<Arc<GlobSet>>,
```
Comment (phillipleblanc) [inline crates/data_components/src/github.rs]:
```suggestion
        include_pattern: Option<Arc<GlobSet>>,
```

**Review by phillipleblanc — DISMISSED** [review]

File: crates/data_components/src/github.rs (hunk 3 above)
Comment (sgrebnov) [reply crates/data_components/src/github.rs]: Updated, thank you

**Review by Sevenannn — APPROVED** [review]

**Pull request merged**